pubsub/tests/unit/pubsub_v1/publisher/batch/test_thread.py

```diff
@@ -361,6 +369,51 @@ def test_publish_exceed_max_messages():
         assert batch._futures == futures
 
 
+@mock.patch.object(thread, "_SERVER_PUBLISH_MAX_BYTES", 1000)
+def test_publish_single_message_size_exceeds_server_size_limit():
+    batch = create_batch(
+        topic="topic_foo",
+        max_messages=1000,
+        max_bytes=1000 * 1000,  # way larger than (mocked) server side limit
+    )
+
+    big_message = types.PubsubMessage(data=b"x" * 984)
+
+    request_size = types.PublishRequest(
+        topic="topic_foo", messages=[big_message]
+    ).ByteSize()
+    assert request_size == 1001  # sanity check, just above the (mocked) server limit
+
+    with pytest.raises(ValueError) as error:
+        batch.publish(big_message)
+
+    error_msg = str(error)
+    assert "too large" in error_msg
+    assert "1000 bytes" in error_msg
```

Comment:
Checking for specific strings in error messages is fragile and can lead to unexpected test flake during future refactors. Catching the error should be enough, no?

Reply:
Since `ValueError` is somewhat general and can, in principle, be raised elsewhere, too, the test (heuristically) tries to check that it is indeed "our" ValueError, and that its message is informative.

I agree its not ideal, hence using very short snippets in the error message, but I still see value in it. I can still remove that part, though, if the concern over future flakes is strong?

Reply:
A pattern I like to use is to make a big hierarchy of exception types and throw/catch based on type. It's more amenable to refactoring and makes the code a little more self documenting (IMO).

Something like
```python
class TooLargeMessageError(ValueError):
   pass

with pytest.raises(TooLargeMessageError) as e:
   # do a thing
```

Reply:
I like that pattern, too, will incorporate it and then we can ditch the assertions about the error message in the test.

Reply:
👍